jgrapht-ext/src/main/java/org/jgrapht/ext/DOTImporter.java

```diff
@@ -256,37 +270,57 @@ private int processHeader(
         }
 
         char current = input.charAt(position);
-        sectionBuffer.append(current);
+
         if (current == '{') {
             // reached the end of the header. Validate it.
 
             String[] headerParts = sectionBuffer.toString().split(" ", 4);
-            if (headerParts.length < 3) {
+            // only graph/digraph keywords are required by the language specifications
+            if (headerParts.length < 1) {
                 throw new ImportException("Not enough parts in header");
             }
 
             int i = 0;
             if (graph instanceof AbstractBaseGraph
                 && ((AbstractBaseGraph<V, E>) graph).isAllowingMultipleEdges()
-                && headerParts[i].equals("strict"))
+                && headerParts[i].toLowerCase()
+                    .equals(DOTExporter.DONT_ALLOW_MULTIPLE_EDGES_KEYWORD))
             {
-                throw new ImportException("graph defines strict but Multigraph given.");
-            } else if (headerParts[i].equals("strict")) {
+                throw new ImportException("graph defines "
+                        + DOTExporter.DONT_ALLOW_MULTIPLE_EDGES_KEYWORD + " but Multigraph given.");
+            } else if (headerParts[i].toLowerCase()
+                    .equals(DOTExporter.DONT_ALLOW_MULTIPLE_EDGES_KEYWORD)) {
                 i = i + 1;
             }
 
-            if ((graph instanceof DirectedGraph) && headerParts[i].equals("graph")) {
+            if ((graph instanceof DirectedGraph)
+                    && headerParts[i].toLowerCase().equals(DOTExporter.UNDIRECTED_GRAPH_KEYWORD)) {
                 throw new ImportException(
                     "input asks for undirected graph and directed graph provided.");
-            } else if (!(graph instanceof DirectedGraph) && headerParts[i].equals("digraph")) {
+            } else if (!(graph instanceof DirectedGraph)
+                    && headerParts[i].equals(DOTExporter.DIRECTED_GRAPH_KEYWORD)) {
                 throw new ImportException(
                     "input asks for directed graph but undirected graph provided.");
-            } else if (!headerParts[i].equals("graph") && !headerParts[i].equals("digraph")) {
-                throw new ImportException("unknown graph type");
+            } else if (!headerParts[i].toLowerCase().equals(DOTExporter.UNDIRECTED_GRAPH_KEYWORD)
+                    && !headerParts[i].toLowerCase().equals(DOTExporter.DIRECTED_GRAPH_KEYWORD)) {
+                throw new ImportException("unknown graph type: " + headerParts[i]);
+            }
+
+            // only parse the graph ID if we will update the components and there is a provided header
+            if (graphUpdater != null && headerParts.length > ++i) {
```

Comment:
Although I changed it, I did it like that because parsing the graph ID is not necessary if the updater was not provided. But I think that it is better to parse it even if there is no updater, to throw an error if the header is not well formed.